subsys/kafka/src/main/java/org/commonjava/indy/subsys/kafka/handler/RepoServiceEventHandler.java

```diff
@@ -181,14 +182,19 @@ private Map<EventStoreKey, ArtifactStore> getStoreMap( DefualtIndyStoreEvent sto
         final Map<EventStoreKey, ArtifactStore> storeMap = new HashMap<>();
         for ( EventStoreKey eventStoreKey : storeEvent.getKeys() )
         {
-            StoreKey storeKey = convertToStoreKey( eventStoreKey );
-            ArtifactStore store = storeDataManager.getArtifactStore( storeKey );
-            if ( store == null )
+            if ( storeDataManager instanceof ServiceStoreDataManager )
             {
-                logger.error( "Failed to fetch store {} through data manager.", storeKey );
-                continue;
+                logger.info( "Get store through store data manager {} force query.",
+                             storeDataManager.getClass().getName() );
+                StoreKey storeKey = convertToStoreKey( eventStoreKey );
+                ArtifactStore store = ( (ServiceStoreDataManager) storeDataManager ).getArtifactStore( storeKey, true );
+                if ( store == null )
+                {
+                    logger.error( "Failed to fetch store {} through data manager.", storeKey );
+                    continue;
+                }
+                storeMap.put( eventStoreKey, store );
             }
```

Comment:
Recommend to add "else" branch to cover other StoreDataManager for safety( even if they may not be called here)

Reply:
@ligangty If it's not service store data manager here, will just return an EMPTY storeMap, I don't find any action needs to do for the other data managers, WDYT?

Reply:
Or maybe a better way is to add an error log? (for protection)